src/Particle/LongRange/StructFact.cpp

```diff
@@ -69,19 +73,32 @@ void StructFact::resize(int ns, int nptcl, int nkpts)
 }
 
 
-void StructFact::UpdateAllPart(ParticleSet& P) { FillRhok(P); }
+void StructFact::UpdateAllPart(const ParticleSet& P)
+{
+  ScopedTimer local(update_all_timer_);
+  FillRhok(P);
+}
+
+void StructFact::mw_UpdateAllPart(const RefVectorWithLeader<StructFact>& sk_list,
```

Comment:
Why isn't this mw_FillRhok or whatever name would actually give a clue up the stack what was happening.

Reply:
The plan is to keep updateAllPart and mw_updateAllPart and remove FillRhok.